Utilities/Cleanup-AzureStorageAccounts.ps1

```diff
@@ -0,0 +1,260 @@
+
+    # Comma separated Azure regions. If empty, All regions in current subscriptions will be selected.
+    [string] $AzureRegions,
+
+    # Comma separated values of exact VHD names to clean.
+    [String] $VHDNames,
+
+    # Command separated VHD name patterns to clean.
+    [String] $Patterns = "LISAv2,-osdisk.vhd,AUTOBUILT",
+
+    # Cleanup Age (in Days)
+    [int] $CleanupAgeInDays = 7,
+
+    # Switch to enable logs for skipped files. (Makes execution slower).
+    [switch] $ShowSkippedLogs,
+
+    # Switch to enable dry run mode.
+    [switch] $DryRun
+)
+
+#region Authenticate the powershell session.
+try {
+    if ($SecretFilePath) {
+        .\Utilities\AddAzureRmAccountFromSecretsFile.ps1 -customSecretsFilePath $SecretFilePath
+        $SubID = ([xml](Get-Content $SecretFilePath)).Secrets.SubscriptionID
+    } else {
+        Write-LogInfo "Secret File not provided. Checking subscriptions..."
+        $Subscriptions = Get-AzSubscription
+        if ($Subscriptions) {
+            $SelectedSubscription = Select-AzSubscription -Subscription $Subscriptions[0].Id
+            Write-LogInfo "Selected $($SelectedSubscription.Name)"
+            $SubID = $Subscriptions[0].Id
+        } else {
+            Write-LogErr "Powershell Session is not authenticated / User doesn't have access to any subscription. Exiting."
+            exit 1
+        }
+    }
+    #endregion
+    Function Check-Pattern ($DiskName, $Patterns) {
+        $retValue = $false
+        if ( $Patterns ) {
+            $Patterns = $Patterns.Split(",")
+            foreach ($item in $Patterns) {
+                if ($DiskName -imatch $item) {
+                    $retValue = $true
+                    break;
+                }
+            }
+        }
+        return $retValue
+    }
+
+    Function Check-FileAge ($Disk) {
+        $DiskCreationTime = $Disk.ICloudBlob.Properties.Created.UtcDateTime
+        $CurrentTime = (Get-Date).ToUniversalTime()
+        $FileAge = ($CurrentTime - $DiskCreationTime).Days
+        return $FileAge
+    }
+
+    Function New-DiskObject ($disk) {
+        $DiskObject = New-Object -TypeName psobject
+        $DiskObject | Add-Member -MemberType NoteProperty -Name Subscription -Value $SubID
+        $DiskObject | Add-Member -MemberType NoteProperty -Name DiskName -Value $disk.Name
+        $size = [math]::Round($disk.Length / 1GB, 3)
+        $DiskObject | Add-Member -MemberType NoteProperty -Name Size -Value $size
+        $DiskObject | Add-Member -MemberType NoteProperty -Name Age -Value $null
+        $DiskObject | Add-Member -MemberType NoteProperty -Name LockStatus -Value $disk.ICloudBlob.Properties.LeaseStatus
+        $DiskObject | Add-Member -MemberType NoteProperty -Name StorageAccount -Value $disk.ICloudBlob.ServiceClient.StorageUri.PrimaryUri.Host.Split(".")[0]
+        $DiskObject | Add-Member -MemberType NoteProperty -Name ResourceGroup -Value $disk.ICloudBlob.ServiceClient.StorageUri.PrimaryUri.Host.Split(".")[0]
```

Comment:
StorageAccount and ResourceGroup are initialized with the same value, I know they will be assigned value later, but a little confusion here.

Reply:
Fixed